tests/boards/tenstorrent/tt_blackhole/clock_control/src/main.c

```diff
@@ -33,52 +33,46 @@ ZTEST(clock_control_rate, test_set_rate_gddr)
 {
 	const struct device *pll = DEVICE_DT_GET(DT_NODELABEL(pll0));
 	uint32_t new_rate;
-	const uint32_t target_rate = 700;
 	int ret;
 
 	clock_control_subsys_t gddr_subsys =
 		(clock_control_subsys_t)CLOCK_CONTROL_TT_BH_CLOCK_GDDRMEMCLK;
 
-	ret = clock_control_set_rate(pll, gddr_subsys, (clock_control_subsys_rate_t)target_rate);
-	zassert_ok(ret, "set_rate for GDDR failed with %d", ret);
+	for (uint32_t target_rate = 750; target_rate <= 1000; target_rate += 50) {
```

Comment:
Would be good to use a Kconfig file local to the testsuite to provide config variables for the minimum value, maximum value, and increment.